### PR TITLE
Fixes #295

### LINKED ISSUE #295: make model explicit when running juju-cli commands or API calls
utilize `juju -m [controller:]model` when executing juju cli commands. Currently we switch to the model we want and then run juju. This has the effect of killing any other existing conjure-up that may be running in seperate model.

As for the API we always switch to the known current model and login to the api, we need to login to the API without switching models.

### PR DESCRIPTION
**Pull request by battlemidget**
Gives the ability to run multiple conjure-up installs, allows the user
to switch controllers/models without interupting the current conjure-up
deployment, basically isolating each deployment.

Signed-off-by: Adam Stokes <battlemidget@users.noreply.github.com>

### CODE REVIEW

File: conjureup/juju.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 def requires_login(f):
     def _decorator(*args, **kwargs):
         if not this.IS_AUTHENTICATED:
-            login()
+            login(True)
```
Comment (mikemccracken) [inline conjureup/juju.py]: It's more readable to put the keyword in here, like `login(force=True)` , then you know what the True is for

File: conjureup/juju.py [hunk 2]
```diff
@@ -559,16 +513,17 @@ def get_accounts():
     raise Exception("Unable to find accounts")
 
 
-def model_by_owner(user):
+def model_by_owner(controller, user):
```
Comment (mikemccracken) [inline conjureup/juju.py]: it looks like model_by_owner isn't used anywhere, and it's a little odd - it'll just return the arbitrary first model owned by 'user'.

What do you think about just getting rid of it?

File: test/test_controllers_clouds_gui.py [hunk 3]
```diff
@@ -91,7 +91,9 @@ def test_finish_w_controller(self):
         self.mock_gcc.return_value = 'testcontroller'
         self.controller.finish('testcloud')
         self.mock_juju.assert_has_calls([
-            call.switch_controller('testcontroller')])
+            ANY,
+            ANY,
```
Comment (mikemccracken) [inline test/test_controllers_clouds_gui.py]: What are the first two calls here? if possible we should test them, or else just avoid asserting anything about them, and use assert_called_once_with or something

File: conjureup/juju.py (hunk 1 above)
Comment (battlemidget) [inline conjureup/juju.py]: Fixed

File: conjureup/juju.py (hunk 2 above)
Comment (battlemidget) [inline conjureup/juju.py]: removed i did a quick grep and verified it wasn't ued anywhere

File: test/test_controllers_clouds_gui.py (hunk 3 above)
Comment (battlemidget) [inline test/test_controllers_clouds_gui.py]: fixed the test as well

**Pull request merged**